### PR TITLE
AV-13: Add Prometheus metrics for relevant libp2p primitives

### PR DESCRIPTION
**Pull request by momosh-ethernal**

### CODE REVIEW

**Review by aterentic-ethernal — APPROVED** [review]

File: src/telemetry/http_service.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+impl MetricService {
+	fn get_reg(&mut self) -> SharedRegistry {
+		Arc::clone(&self.reg)
+	}
+
+	fn respond_with_metrics(&mut self) -> Response<Body> {
+		let mut encoded: Vec<u8> = Vec::new();
+		let reg = self.get_reg();
+
+		encode(&mut encoded, &reg.lock().unwrap()).unwrap();
```
Comment (sh3ll3x3c) [inline src/telemetry/http_service.rs:26]: We should prob do some error handling here...

File: src/telemetry/http_service.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		let metrics_content_type = "application/openmetrics-text;charset=utf-8;version=1.0.0";
+		Response::builder()
+			.status(StatusCode::OK)
+			.header(hyper::header::CONTENT_TYPE, metrics_content_type)
+			.body(Body::from(encoded))
+			.unwrap()
```
Comment (sh3ll3x3c) [inline src/telemetry/http_service.rs:32]: We prob shouldn't panic here...

File: src/telemetry/http_service.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	fn respond_with_404_not_found(&mut self) -> Response<Body> {
+		Response::builder()
+			.status(StatusCode::NOT_FOUND)
+			.body(Body::from("Not found. Try localhost:[port]/metrics"))
+			.unwrap()
```
Comment (sh3ll3x3c) [inline src/telemetry/http_service.rs:39]: Also needs to be handled

File: src/telemetry/http_service.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	fn get_reg(&mut self) -> SharedRegistry {
+		Arc::clone(&self.reg)
+	}
+
+	fn respond_with_metrics(&mut self) -> Response<Body> {
+		let mut encoded: Vec<u8> = Vec::new();
+		let reg = self.get_reg();
+
+		encode(&mut encoded, &reg.lock().unwrap()).unwrap();
+		let metrics_content_type = "application/openmetrics-text;charset=utf-8;version=1.0.0";
```
Comment (sh3ll3x3c) [inline src/telemetry/http_service.rs:27]: We could try to serve just a simple json for ease of use on the `/metric` endpoint, and create a separate endpoint for OM. This content type triggers file download in browsers, which is not a desired outcome for further prometheus integration (ie. Datadog).

**Review by luka-ethernal — APPROVED** [review]

**Pull request merged**